app/demo-opentitan-earlgrey/src/main.rs

```diff
@@ -0,0 +1,37 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+#![no_main]
+
+extern crate panic_halt;
+extern crate riscv_rt;
+
+use riscv_rt::entry;
+
+#[entry]
+fn main() -> ! {
+    // The opentitan requires enabling the timer and interrupts
+    // for mtime and mtimecmp to behave as expected
+    // See https://docs.opentitan.org/hw/ip/rv_timer/doc/
+    const OT_TIMER_CTRL: u32 = 0x4010_0004;
+    const OT_TIMER_INTR: u32 = 0x4010_0100;
+
+    // Enable mtime to count up
+    let timer = OT_TIMER_CTRL as *mut u32;
+    unsafe {
+        *timer |= 1;
+    };
+
+    // Enable interrupts when mtime passes mtimecmp
+    let intr = OT_TIMER_INTR as *mut u32;
+    unsafe {
+        *intr |= 1;
+    };
+
+    // TODO this will need to change when run on actual hardware
+    const MTIMECMP: u32 = 1000;
```

Comment:
Why is this named `MTIMECMP`? This argument to start_kernel is supposed to be `tick_divisor`.

Reply:
From what I understand, this is what the `mtimecmp` register is set to, and the kernel periodicially resets `mtime` back to 0.

Reply:
I think that is an internal implementation detail of the rv32 architecture ;). The function signature expects `tick_divisor` i.e. the divisor to convert machine ticks into kernel ticks. It just seems a little weird to use MTIMECMP directly here.